File: Correctness/StringFormatMatches/src/main/java/net/twisterrob/lints/correctness/StringFormatMatchesViolation.java
package net.twisterrob.lints.correctness;

import android.content.res.Resources;

import net.twisterrob.lints.correctness.StringFormatMatches.R;

public class StringFormatMatchesViolation {

	String format(Resources res) {
		return ""
				+ res.getString(R.string.invalid, "text")
				+ res.getString(R.string.suspicious, 42)
				+ res.getString(R.string.mismatched_count, "text");
	}
}

File: Correctness/StringFormatMatches/src/main/java/net/twisterrob/lints/correctness/StringFormatMatchesViolation.kt
package net.twisterrob.lints.correctness

import android.content.res.Resources
import net.twisterrob.lints.correctness.StringFormatMatches.R

fun format(res: Resources) = listOf(
	res.getString(R.string.invalid, "text"),
	res.getString(R.string.suspicious, 42),
	res.getString(R.string.mismatched_count, "text")
)
